apollo-http-cache/src/main/kotlin/com/apollographql/apollo3/cache/http/DiskLruHttpCache.kt

```diff
@@ -104,6 +137,51 @@ class DiskLruHttpCache(private val fileSystem: FileSystem, private val directory
     }
   }
 
+  /**
+   * A [BufferedSource] that writes to the given cache sink as it is read.
+   */
+  private class WriteToCacheSource(
+      private val originalBody: Source,
+      private val cacheSink: Sink,
+      private val cacheEditor: DiskLruCache.Editor,
+  ) : Source by originalBody {
```

Comment:
[Interface delegation is one of Andrey Breslav regrets in Kotlin design](https://discuss.kotlinlang.org/t/what-is-wrong-about-implementation-by-delegation/19873) so I'd prefer avoiding it. 

Especially in that case, looks like the only thing we need to proxy is `timeout`?